core/arch/arm/kernel/boot.c

```diff
@@ -1338,6 +1350,17 @@ void __weak boot_init_primary_late(unsigned long fdt)
 		DMSG("Memory tagging %s",
 		     memtag_is_enabled() ?  "enabled" : "disabled");
 
+	/* Check if platform needs NMFI workaround */
+#if CFG_TEE_CORE_LOG_LEVEL >= TRACE_DEBUG
+	if (cpu_nmfi_enabled())	{
+		if (!IS_ENABLED(CFG_CORE_WORKAROUND_ARM_NMFI))
+			IMSG("WARNING: This ARM core has NMFI enabled, please apply workaround!");
+	} else {
+		if (IS_ENABLED(CFG_CORE_WORKAROUND_ARM_NMFI))
+			IMSG("WARNING: This ARM core does not have NMFI enabled, no need for workaround.");
```

Comment:
Nit: preferred style is to omit full stop (`.`) when there is single sentence in log messages. Same in comments: `/* One sentence, no full stop */` but `/* Two sentences. Full stops needed. */`.

Reply:
OK. I am not native English speaker -- there are two parts in here -- but I assume this is what you are looking for:
```
IMSG("WARNING: This ARM core does not have NMFI enabled, no need for workaround");
```
eg. dot removed from the end. 

Will update as recommended.